pkg/relayer/proxy/async.go

```diff
@@ -49,12 +50,29 @@ func (server *relayMinerHTTPServer) handleAsyncConnection(
 	if !ok {
 		return ErrRelayerProxyServiceEndpointNotHandled
 	}
-	supplierServiceConfig := supplierConfig.ServiceConfig
+
+	// Initialize the service config to the default service config.
+	serviceConfig := supplierConfig.DefaultServiceConfig
+
+	if serviceConfig == nil {
+		return ErrRelayerProxyServiceEndpointNotHandled.Wrapf(
+			"service %q not configured",
+			serviceId,
+		)
+	}
+
+	// If the RPC-Type header is set, use the RPC type-specific service config.
+	rpcType := config.RPCType(request.Header.Get("RPC-Type"))
```

Comment:
This is where it's a bit foggy. Is there a case where we've already determined that it's a websockets connection but still need to look for the `RPCType`?

I believe that at some point `websocket` should only be an `RPCType` instead of a dedicated service.

Reply:
Based on our discussion in Discord I've updated the PR to use the `Rpc-Type: websocket` header to determine whether to use the websocket connection method vs the regular sync method. 

Please see this commit for details: https://github.com/pokt-network/poktroll/pull/1633/commits/6e6e462d4e89068bf0b80780ada529765f0b1a3f